udp.go

```diff
@@ -4,6 +4,11 @@ package dns
 
 import (
 	"net"
+	"os"
+	"syscall"
+
+	"golang.org/x/net/ipv4"
```

Comment:
We only use the std lib, so this is a *big* change and will mean we can't merge it

Reply:
Didn't realize that restriction. I'll work on removing that.